InternTrack.Core.Api.Tests.Unit/Services/Foundations/Interns/InternServiceTests.Validations.Modify.cs

```diff
@@ -0,0 +1,380 @@
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Theory]
+        [MemberData(nameof(MinutesBeforeOrAfter))]
+        public async Task ShouldThrowValidationExceptionOnModifyIfUpdatedDateIsNotRecentAndLogItAsync(
+            int minutesBeforeOrAfter)
+        {
+            // given
+            DateTimeOffset randomDateTime = GetRandomDateTime();
+            Intern randomIntern = CreateRandomIntern(randomDateTime.AddMinutes(minutesBeforeOrAfter));
+            Intern invalidIntern = randomIntern;
+            invalidIntern.UpdatedBy = invalidIntern.CreatedBy;
+            invalidIntern.UpdatedDate = invalidIntern.UpdatedDate.AddMinutes(minutesBeforeOrAfter);
+
+            var invalidInternException =
+                new InvalidInternException();
+
+            invalidInternException.AddData(
+                key: nameof(Intern.UpdatedDate),
+                values: "Date is not recent");
+
+            var expectedInternValidationException =
+                new InternValidationException(invalidInternException);
+
+            this.dateTimeBrokerMock.Setup(broker =>
+                broker.GetCurrentDateTimeOffset())
+                    .Returns(randomDateTime);
+
+            this.storageBrokerMock.Setup(broker =>
+                broker.SelectInternByIdAsync(invalidIntern.Id))
+                    .ReturnsAsync(invalidIntern);
+
+            // when
+            ValueTask<Intern> modifyInternTask =
+                this.internService.ModifyInternAsync(invalidIntern);
+
+            InternValidationException actualInternValidationException =
+                await Assert.ThrowsAsync<InternValidationException>(
+                    modifyInternTask.AsTask);
+
+            // then
+            actualInternValidationException.Should().BeEquivalentTo(
+                expectedInternValidationException);
+
+            this.dateTimeBrokerMock.Verify(broker =>
+                 broker.GetCurrentDateTimeOffset(),
+                     Times.Once);
+
+            this.loggingBrokerMock.Verify(broker =>
+                broker.LogError(It.Is(SameValidationExceptionAs(
+                    expectedInternValidationException))),
+                        Times.Once);
+
+            this.storageBrokerMock.Verify(broker =>
+               broker.SelectInternByIdAsync(randomIntern.Id),
+                   Times.Never);
+
+            this.dateTimeBrokerMock.VerifyNoOtherCalls();
+            this.loggingBrokerMock.VerifyNoOtherCalls();
+            this.storageBrokerMock.VerifyNoOtherCalls();
+        }
+
+        [Fact]
+        public async Task ShouldThrowValidationExceptionOnModifyIfInternDoesntExistAndLogItAsync()
+        {
+            DateTimeOffset randomDateTimeOffset = GetRandomDateTimeOffset();
+            Intern nonExistentIntern = CreateRandomIntern();
+            nonExistentIntern.UpdatedDate = randomDateTimeOffset;
```

Comment:
Pass random Date as parameter. You might have to create method that handles the random object creation specific to Modify use-cases. That makes CreatedDate earlier than UpdatedDate